domain/upgrade/state/type.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
```

Comment:
in `externalcontroller` we named this file as `types.go`, I don't have any opinion, but it'd be better if we have a standard naming for these files

Reply:
Oops I think this was a typo